dist/PaginationBoxView.js

```diff
@@ -39,41 +39,58 @@ var PaginationBoxView = React.createClass({
     };
   },
 
-  handlePageSelected: function handlePageSelected(selected, event) {
-    event.preventDefault();
+  callClickCallback: function callClickCallback() {
+    if (typeof this.props.clickCallback !== 'undefined' && typeof this.props.clickCallback === 'function') {
+      this.props.clickCallback({ selected: this.state.selected });
+    }
+  },
 
-    if (this.state.selected === selected) {
-      return;
-    }this.setState({ selected: selected });
+  handlePageSelected: function handlePageSelected(selected, event) {
+    if (typeof event !== 'undefined') {
+      event.preventDefault();
+    }
 
-    if (typeof this.props.clickCallback !== 'undefined' && typeof this.props.clickCallback === 'function') {
-      this.props.clickCallback({ selected: selected });
+    var needToUpdate = this.state.selected !== selected;
+    if (needToUpdate) {
+      this.setState({ selected: selected }, this.callClickCallback);
+    } else {
+      this.callClickCallback();
```

Comment:
I understand why you want to use setState's callback here but I think we don't need to call the 'callClickCallback' function if the variable 'selected' equals the previous state.selected.